rmfakecloud: 0.0.21 -> 0.0.23

martinetd opened the pull request:
## Things done

- fix regression in 0.0.21 for old remarkable tablet versions (see https://github.com/ddvk/rmfakecloud/issues/330 )
- fix build with new UI... tested it appears to work ok

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

martinetd committed:
rmfakecloud: 0.0.21 -> 0.0.23

0.0.22
- fix regression in 0.0.21 for old remarkable tablet versions
(see https://github.com/ddvk/rmfakecloud/issues/330 )
- fix build with new UI...

0.0.23
- fix other regression with 0.0.22...

dotlambda commented on pkgs/by-name/rm/rmfakecloud/webui.nix:
```diff
@@ -1,37 +1,46 @@
   meta = with lib; {
     description = "Only the webui files for rmfakecloud";
     homepage = "https://ddvk.github.io/rmfakecloud/";
     license = licenses.agpl3Only;
```
`maintainers` missing

martinetd replied:
That derivation cannot be eused as is so I'd ather remove all the meta fluff instead (it's a build dep of rmfakecloud itself which has maintainers)

Is that ok?

dotlambda replied:
No, you should rather inherit the meta attributes that the two share from rmfakecloud.

martinetd replied:
That'd inherit the mainProgram attribute, I'm not sure we want that.. Well wouldn't matter if nobody uses it except for cp I guess.

Anyway, I've managed to build the go module with pnpm inside after trying again (not sure why it didn't find the deps last time I tried), it's gone.
Hopefully that'll allow the update bot to build updates automatically next time, I was thinking it had troubles updating hashes in two separate files in lockstep.

dotlambda commented on pkgs/by-name/rm/rmfakecloud/package.nix:
```diff
@@ -9,13 +9,13 @@
 
   src = fetchFromGitHub {
     owner = "ddvk";
     repo = pname;
```
#277994

martinetd replied:
fixed

martinetd committed:
rmfakecloud: build the webui together with the app

This avoids maintaining an artificial derivation for the web UI,
and hopefully will allow the auto-update bot to make PRs for further
updates

martinetd commented:
fudge, ofborg is getting the same error I was initially getting with `Error: 'pnpmDeps' must be set when using pnpmConfigHook.`

So it must only have been working locally because I had my store pre-populated with something from the successful build... I'll just drop the last commit for now because this is @$#@%@ supposed to be a bug fix release and I've already sunk way too much time into it than I should have.
If someone wants to look the "bring pnpmDeps in the go module" commit is available at https://github.com/martinetd/nixpkgs/commits/rmfakecloud_gopnpm/

martinetd commented:
(This works with workaround given in #358844 and (almost) makes the update script work, so going with the "merge everything in one file" version)

malte-v commented:
Is this ready to be merged?

martinetd commented:
Yes, aside of the detail that the go module indirectly depends on the pnpm one (meaning autoupgrade won't work) this is all in working order as far as I'm concerned
